contracts/bonding/BondingManager.sol

```diff
@@ -165,41 +165,76 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         whenSystemNotPaused
         currentRoundInitialized
     {
-        // Only callable if current round is not locked
-        require(!roundsManager().currentRoundLocked());
-        // Block reward cut must be a valid percentage
-        require(MathUtils.validPerc(_blockRewardCut));
-        // Fee share must be a valid percentage
-        require(MathUtils.validPerc(_feeShare));
-
+        Transcoder storage t = transcoders[msg.sender];
         Delegator storage del = delegators[msg.sender];
 
-        // Must have a non-zero amount bonded to self
-        require(del.delegateAddress == msg.sender && del.bondedAmount > 0);
+        if (roundsManager().currentRoundLocked()) {
+            // If it is the lock period of the current round
+            // the lowest price previously set by any transcoder
+            // becomes the price floor and the caller can lower its
+            // own price to a point greater than or equal to the price floor
+
+            // Caller must already be a registered transcoder
+            require(transcoderStatus(msg.sender) == TranscoderStatus.Registered);
+
+            // Iterate through the transcoder pool to find the price floor
+            // Since the caller must be a registered transcoder, the transcoder pool size will always at least be 1
+            // Thus, we can safely set the initial price floor to be the pendingPricePerSegment of the first
+            // transcoder in the pool
+            address currentTranscoder = transcoderPool.getFirst();
+            uint256 priceFloor = transcoders[currentTranscoder].pendingPricePerSegment;
+            for (uint256 i = 0; i < transcoderPool.getSize(); i++) {
```

Comment:
I wonder if it's worth it to track the priceFloor per round (if it's already been calculated) in a data structure so that we don't have to loop through to determine it every time.

Reply:
The tricky part with tracking the priceFloor is computing the new priceFloor if the transcoder with the previous priceFloor resigns. I think we would have to maintain a data structure (likely a sorted one) that can always return the correct priceFloor even if transcoders enter/leave the transcoder pool during a round which would require a bunch of additional storage writes to handle this case

Reply:
Got it. Ok let's leave this for now. I can envision adding additional accounting to handle this case, such as a bool mapping that returns false if the floor isn't locked for the round, and then sets this to false if a transcoder resigns. But I also don't like the idea of adding all this special purpose accounting all over the place. I'd rather keep it simple than optimize for gas savings at this point.